### ISSUE TITLE
The 'pod install' command creates a group in the project navigator with '..' name

### ISSUE THREAD

**Issue by wojciechczerski** [CocoaPods/CocoaPods#6979]
🌈

* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

1. Run `pod lib create MyLibrary`
2. Provide the following answers to the library template: 
```
What language do you want to use?? [ Swift / ObjC ]
 > Swift

Would you like to include a demo application with your library? [ Yes / No ]
 > Yes

Which testing frameworks will you use? [ Quick / None ]
 > None

Would you like to do view based testing? [ Yes / No ]
 > No
```
3. Run `pod install`
4. Open the _Classes_ directory in the project - it should contain only one `ReplaceMe.swift` file
5. Change the structure of the _Classes_ directory - its direct children should be at least **2 folders** and **NO** `*.swift` files:

![bildschirmfoto 2017-08-23 um 22 10 37](https://user-images.githubusercontent.com/1027098/29636176-f1d1d5b4-884f-11e7-8493-86baf901b1e9.png)

6. Run `pod install`

## What did you expect to happen?

I expect to see the groups created in the project navigator to reflect the structure of the _Classes_ directory:

![bildschirmfoto 2017-08-23 um 22 17 53](https://user-images.githubusercontent.com/1027098/29637255-db73927c-8853-11e7-958e-da5961a996e9.png)

## What happened instead?

An incorrect group with name `..` is created:

![bildschirmfoto 2017-08-23 um 22 16 52](https://user-images.githubusercontent.com/1027098/29637328-10832522-8854-11e7-85e2-652c7e7dd36f.png)

### Stack

```
   CocoaPods : 1.3.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.6 (16G29)
       Xcode : 8.3.3 (8E3004b)
         Git : git version 2.11.0 (Apple Git-81)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 4f31589b4871a8603d3ff84610f37da26cfe3983
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-clean       : 0.0.1
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.2.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
use_frameworks!

target 'MyLibrary_Example' do
  pod 'MyLibrary', :path => '../'

  target 'MyLibrary_Tests' do
    inherit! :search_paths

    
  end
end
```

**Comment by orta** [comment]
Interesting, that does make sense in that the closest folder relative to it all is back a folder. 

Might be worth either updating the CocoaPods project generation to ignore cases like this, or we just say that's the intended behaviour.

**Comment by wojciechczerski** [comment]
Thanks for the answer! I did some further search on this issue and it looks that it was introduced in version **1.3.0.beta.3**. According to the release notes it must be a side effect of the feature #6814 (_Truncate shared parent folders in Development Pods to the nearest ancestor_).

This new feature is very good and I really hope it will be possible to improve it so that it can handle the _edge case_ I've described.
